store/src/java/com/zimbra/qa/unittest/TestFolderACLCache.java

```diff
@@ -16,150 +16,182 @@
+     *
+     * 2. Test the case when memcached is configured/running.
      *    zmprov mcf zimbraMemcachedClientServerList 'localhost:11211'
      *    (restart server)
      *    /opt/zimbra/memcached/bin/memcached -vv
      *    ==> run the test
      */
-    
-    String OWNER_ACCT_ID;
-    Account USER1;
-    Account USER2;
-    Account USER3;
-    
+
+    Account ownerAcct = null;
+    Account USER2 = null;
+    Account USER3 = null;
+    String OWNER_ACCT_ID = null;
+    static List<Account> accts = Lists.newArrayList();
+    Provisioning prov = Provisioning.getInstance();
+
     int INBOX_FID;
     int SUB1_FID;
     int SUB2_FID;
     int SUB3_FID;
-    
+
+    @Before
     public void setUp() throws Exception {
-        
-        Provisioning prov = Provisioning.getInstance();
-        
-        Account ownerAcct = prov.get(AccountBy.name, "user1");
+        testName = testInfo.getMethodName();
+        userNamePrefix = "testfolderaclcache-" + testName + "-user-";
+
+        ownerAcct = createAcct(userNamePrefix + "1");
+        USER2 = createAcct(userNamePrefix + "haswriteaccess");
+        USER3 = createAcct(userNamePrefix + "hasreadwriteaccess");
         ZMailbox ownerMbx = TestUtil.getZMailbox(ownerAcct.getName());
-        
+
         ZFolder inbox = ownerMbx.getFolderByPath("inbox");
-        
+
         String sub1Path = "/inbox/sub1";
         ZFolder sub1 = ownerMbx.getFolderByPath(sub1Path);
         if (sub1 == null) {
             sub1 = TestUtil.createFolder(ownerMbx, sub1Path);
-            ownerMbx.modifyFolderGrant(sub1.getId(), ZGrant.GranteeType.usr, "user2", "w", null);
-            ownerMbx.modifyFolderGrant(sub1.getId(), ZGrant.GranteeType.usr, "user3", "rw", null);
-        } 
-        
+            ownerMbx.modifyFolderGrant(sub1.getId(), ZGrant.GranteeType.usr, USER2.getName(), "w", null);
+            ownerMbx.modifyFolderGrant(sub1.getId(), ZGrant.GranteeType.usr, USER3.getName(), "rw", null);
+        }
+
         String sub2Path = "/inbox/sub1/sub2";
         ZFolder sub2 = ownerMbx.getFolderByPath(sub2Path);
         if (sub2 == null)
             sub2 = TestUtil.createFolder(ownerMbx, sub2Path);
-        
+
         String sub3Path = "/inbox/sub3";
         ZFolder sub3 = ownerMbx.getFolderByPath(sub3Path);
         if (sub3 == null) {
             sub3 = TestUtil.createFolder(ownerMbx, sub3Path);
             ownerMbx.modifyFolderGrant(sub3.getId(), ZGrant.GranteeType.pub, null, "r", null);
         }
-        
+
         OWNER_ACCT_ID = ownerAcct.getId();
-        USER1 = prov.get(AccountBy.name, "user1");
-        USER2 = prov.get(AccountBy.name, "user2");
-        USER3 = prov.get(AccountBy.name, "user3");
-        
+
         INBOX_FID = Integer.valueOf(inbox.getId());
         SUB1_FID = Integer.valueOf(sub1.getId());
         SUB2_FID = Integer.valueOf(sub2.getId());
         SUB3_FID = Integer.valueOf(sub3.getId());
     }
-    
+
+    @After
+    public void cleanUp() throws Exception {
```

Comment:
My preference is to call cleanup before and after the test in case previous run was interrupted and left garbage behind. IMHO, this makes the test suite more resilient and easier to run repeatedly.